### ISSUE TITLE
Compilation error RHS in `expect` is a throwing expression

### ISSUE THREAD

**Issue by Sajjon** [apple/swift#68753]
### Description

This code works just fine:

# Works
```swift
func zero() throws -> Int { 0 }

@Test
func works_LHS_throws() throws {
	try #expect(zero() == 0)
}
```

# Does not work 1

```swift
@Test
func does_not_work_RHS_throws_outer_try() throws {
	try #expect(0 == zero())
}
```

# Does not work 2

```swift
@Test
func does_not_work_RHS_throws_try_right_of_EQ() throws {
	#expect(0 == try zero())
}
```

# Works
**However**, this works:
Moving the `try` inside of `#expect` makes it work

```swift
@Test
func works_RHS_throws_try_right_of_LHS() throws {
	#expect(try zero() == 0)
}

```

### Expected behavior

I expect to be using a throwing expression in RHS of `try #expect(lhs == throwingRHS())`, just like I can use `try #expect(throwingLHS == rhs)`.

### Actual behavior

Does not compile

### Steps to reproduce

Just try:

```swift
func zero() throws -> Int { 0 }

@Test
func works_LHS_throws() throws {
	try #expect(zero() == 0)
}

@Test
func does_not_work_RHS_throws_outer_try() throws {
	try #expect(0 == zero())
}
```

### swift-testing version/commit hash

d2c955716c80d619828aa839d783ad0204943fac

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx14.0
Darwin VerktygsLadan.local 23.0.0 Darwin Kernel Version 23.0.0: Fri Sep 15 14:41:43 PDT 2023; root:xnu-10002.1.13~1/RELEASE_ARM64_T6000 arm64

**Comment by DougGregor** [comment]
Right, the error is that one cannot have a `try` on the right-hand side:

```
t.swift:4:8: error: 'try' cannot appear to the right of a non-assignment operator
  0 == try zero()
       ^
```

due to the operator-precedence rules. This is correct behavior.

**Comment by grynspan** [comment]
There are other scenarios (specifically `try #expect(zero() == 0)`) that we're interested in where a developer can write a syntactically valid expression that cannot be expanded correctly by swift-testing's macros.
